SimCalorimetry/HcalSimProducers/src/HcalDigitizer.cc

```diff
@@ -420,7 +423,25 @@ void HcalDigitizer::accumulateCaloHits(edm::Handle<std::vector<PCaloHit>> const
 
   if (isZDC) {
     if (zdcgeo && theZDCDigitizer) {
-      theZDCDigitizer->add(*zdcHandle.product(), bunchCrossing, engine);
+      std::vector<PCaloHit> zdcHitsOrig = *zdcHandle.product();
+      std::vector<PCaloHit> zdcHits;
+      zdcHits.reserve(zdcHitsOrig.size());
+      // eliminate bad hits
+      for (unsigned int i = 0; i < zdcHitsOrig.size(); i++) {
+        DetId id(zdcHitsOrig[i].id());
+        HcalZDCDetId hid(id);
+        if (!ztopoP->valid(hid)) {
```

Comment:
@bsunanda , in real run there will be many events. Is it correct to have unlimited printouts of wrong ZDC hits?

Reply:
I shall protect the printout of good ones - but the wrong DetId is an error and should not happen in normal runs. We have similar error printouts for HCAL which do not produce large printouts